### PR TITLE
Fixes for HEMCO diagnostic config file HEMCO_Diagn.rc

### PR DESCRIPTION
**Pull request by lizziel**
### Name and Institution (Required)

Name: Lizzie Lundgren
Institution: Harvard University

### Describe the update

This PR contains two bug fixes:

1. Update GC-Classic and GCHP `HEMCO_Diagn.rc` fullchem templates to use offline MOH biogenic emissions rather than online unless using the benchmark simulation. This corrects a problem where `EmisMOH_Biogenic` was filled with all zeros for non-benchmark fullchem simulations in 14.4.2 since MEGAN is no longer on starting in that version.
2. Auto-set GCHP emissions sources in `HEMCO_Diagn.rc` based on `HEMCO_Config.rc`.  This update impacts only GCHP emissions for which we have both online and offline sources, namely seasalt, soil NOx, biogenic, and dust emissions. Previously `HEMCO_Diagn.rc` always specified offline emissions for use in emissions diagnostics unless using the benchmark simulation. This resulted in empty emissions diagnostics if the user changed from offline to online.

### Expected changes

This is a no-diff update for benchmark simulations. There are expected changes in the emissions diagnostic collection for non-benchmark full chemistry simulations and they are as follows:
- `EmisMOH_Biogenic` will no longer be all zeros if using GC-Classic and GCHP
- GCHP emission diagnostics will no longer be zero if offline emissions are changed to online

### Reference(s)

None

### Related Github Issue

closes https://github.com/geoschem/geos-chem/issues/2382
Also related to https://github.com/geoschem/geos-chem/issues/2319

### CODE REVIEW

**Comment by lizziel** [comment]
I added these changes on top of dev/14.5.0. I can rebase to be on top of dev/no-diff-to-benchmark instead.

File: CHANGELOG.md
Comment (yantosca) [inline CHANGELOG.md:25]: This is a typo.  I can fix it when I resolve the merge conflicts.

File: test/integration/GCClassic/integrationTestCreate.sh
Comment (yantosca) [inline test/integration/GCClassic/integrationTestCreate.sh:294]: FYI, this has already gone into dev/no-diff-to-benchmark.  These updates will conflict with it.

File: test/shared/commonFunctionsForTests.sh
Comment (yantosca) [inline test/shared/commonFunctionsForTests.sh:35]: FYI, this has already gone into dev/no-diff-to-benchmark.  These updates will conflict with it.

**Review by yantosca — CHANGES_REQUESTED** [review]
Thanks @lizziel, this looks OK to merge.  Could you revert the integrationTestCreate.sh and commonFunctionsForTests.sh files?  The GCAP2 int tests are now in dev/no-diff-to-benchmark... I had to make some updates to that code so this could will cause conflicts.

**Comment by lizziel** [comment]
Yes, I am going to rebase on dev/no-diff-to-benchmark so don't merge yet!

**Comment by yantosca** [comment]
No problem Lizzie!  I'll approve so you can rebase.

File: test/integration/GCClassic/integrationTestCreate.sh
Comment (yantosca) [reply test/integration/GCClassic/integrationTestCreate.sh:294]: Approving with the understanding that @lizziel will rebase to dev/no-diff-to-benchmark.

File: test/shared/commonFunctionsForTests.sh
Comment (yantosca) [reply test/shared/commonFunctionsForTests.sh:35]: Approving with the understanding that @lizziel will rebase to dev/no-diff-to-benchmark.

File: test/shared/commonFunctionsForTests.sh
Comment (yantosca) [inline test/shared/commonFunctionsForTests.sh:283]: Approving with the understanding that @lizziel will rebase to dev/no-diff-to-benchmark.

**Review by yantosca — APPROVED** [review]
Thanks @lizziel.  I approve these updates, since you will rebase later.

**Comment by lizziel** [comment]
@yantosca, this is now rebased on dev/no-diff-to-benchmark. I updated the milestone to be 14.4.3.

**Comment by yantosca** [comment]
Thanks @lizziel!  I merged this into dev/no-diff-to-benchmark (locally) and am running integration tests.

**Comment by yantosca** [comment]
All GEOS-Chem Classic integration tests passed:
```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

CodeDir   : 60dd1a6 GEOS-Chem submod update: Merge PR #2413 (Fix unit conv in aerosol sim)
GEOS-Chem : 034507531 Merge PR #2402 (Fixes for HEMCO_Diagn.rc)
HEMCO     : e68ebae HEMCO 3.9.2 release
Cloud-J   : 37a4ece Update release version number and date
HETP      : 2a99b24 Merge pull request #2 from geoschem/bugfix/initialize_local_variables

Using 24 OpenMP threads
Number of execution tests: 30

Submitted as SLURM job: 43548446
==============================================================================

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
All tests were zero-diff w/r/t the previous PR #2413, except:
- TOMAS (parallelization error?)
- APM (parallelization error?)

The following fullchem tests were zero-diff w/r/t PR #2413 (as evidenced by identical Restart files), but the EmisMOH_biogenic field in HEMCO_diagnostics is no longer all zeros:
- fullchem
- fullchem_aciduptake
- fullchem_APM
- fullchem_complexSOA
- fullchem_complexSOA_SVPOA
- fullchem_LuoWd
- fullchem_RRTMG

**Comment by yantosca** [comment]
All GCHP integration tests passed:
```console
==============================================================================
GCHP: Execution Test Results

CodeDir       : 4cadc0d GEOS-Chem submod update: Merge PR #2413 (Fix unit conv in aerosol sim)
MAPL          : 231d53c Merge pull request #36 from geoschem/feature/improve_hflux_regridding
GMAO_Shared   : 4ddb3ec Merge pull request #2 from geoschem/feature/mapl-upgrade
ESMA_cmake    : ad5deba Added ecbuild as a submodule of ESMA_cmake
gFTL-shared   : 4b82492 Merge branch 'upstream_v1.5.0' into feature/v1.5.0
FMS           : 259759d Merge pull request #3 from geoschem/feature/update_gmao_libs
FVdycoreCubed : af42462 Merge PR #8 (Add PLEadv diagnostic for offline advection in GCHP)
geos-chem     : 034507531 Merge PR #2402 (Fixes for HEMCO_Diagn.rc)
HEMCO         : e68ebae HEMCO 3.9.2 release
yaFyaml       : 19afe50 Merge branch 'upstream_v1.0.4' into feature/v1.0.4
pFlogger      : 2c4b724 Merge branch 'upstream_v1.9.1' into feature/v1.9.1
Cloud-J       : 37a4ece Update release version number and date
HETP          : 2a99b24 Merge pull request #2 from geoschem/bugfix/initialize_local_variables

Number of execution tests: 12

Submitted as SLURM job: 43561008
==============================================================================

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
All tests were zero-diff w/r/t the prior PR #2413.  

The fullchem_benchmark and fullchem_alldiags tests showed differences in the Emissions collection, that were caused by the updates to `setCommonRunSettings.sh` added in this PR.

**Pull request merged**
